.github/workflows/test-suite.yml

```diff
@@ -83,6 +88,7 @@ jobs:
     - name: Install make
       run: choco install -y make
     - uses: KyleMayes/install-llvm-action@v1
+      if: env.SELF_HOSTED_RUNNERS == false
```

Comment:
I think we're good with this but just something worth thinking - Is there any benefit of leaving these conditionals out? 

There are a few benefits:
- CI be always the same dep versions on self-hosted / github-hosted runners
- For Rust, the workflow will automatically update to latest version as soon as it's released, it could take an extra 15s or more to install. But then we'll know we need to update our runner image.

Downside:
- it could be a bit slower, but AFAIK `rustup update stable` just returns immediately if we're already on the same version.

Reply:
IIRC the self-hosted runners fail if we run that llvm install action.

Reply:
Same with the Rust update ones.

Reply:
ahh privileges! happy to leave it as it is 👍

Reply:
IMO it would be ideal if we can have the images perform exactly the same. Perhaps we can make an issue to come back for it.

I suspect that we're going to eventually drift into the territory of Github-hosted runners failing, though. If we're not actually testing on the Github-hosted ones they're bound to bit-rot.

Reply:
Yeah, that's a good point - I'm hesitant to make the workflow more complicated, but if we really want to run GitHub-hosted runners, we could still run them on PRs, and use the self-hosted ones on `staging` 😛 , but perhaps something to think about later down the line.